plugin/backend_linux.go

```diff
@@ -294,11 +294,11 @@ func (pm *Manager) Remove(name string, config *types.PluginRmConfig) error {
 	}
 
 	id := p.GetID()
-	pm.pluginStore.Remove(p)
 	pluginDir := filepath.Join(pm.libRoot, id)
 	if err := os.RemoveAll(pluginDir); err != nil {
-		logrus.Warnf("unable to remove %q from plugin remove: %v", pluginDir, err)
+		return errors.Wrap(err, "failed to remove plugin directory")
```

Comment:
question: If we are force removing and RemoveAll fails, shouldn't it keep the plugin in the store so you can see that it is still there / that remove failed (besides the err)? If it's not in the store, doesn't that means the only way to remove it is by manually removing it from disk?